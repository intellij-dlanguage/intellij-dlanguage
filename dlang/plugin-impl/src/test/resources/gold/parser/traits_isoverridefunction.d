import std.stdio;

void main()
{
    int i;
    writeln(__traits(isArithmetic, int));
    writeln(__traits(isArithmetic, i, i+1, int));
    writeln(__traits(isArithmetic));
    writeln(__traits(isArithmetic, int*));
}