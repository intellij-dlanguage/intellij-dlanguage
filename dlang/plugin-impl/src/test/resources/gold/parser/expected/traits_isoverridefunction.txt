D Language File
  DLanguageDeclarationImpl(DECLARATION)
    DLanguageImportDeclarationImpl(IMPORT_DECLARATION)
      PsiElement(DlangTokenType.import)('import')
      SINGLE_IMPORT
        DLanguageIdentifierChainImpl(IDENTIFIER_CHAIN)
          IDENTIFIER
            PsiElement(DlangTokenType.ID)('std')
          PsiElement(DlangTokenType..)('.')
          IDENTIFIER
            PsiElement(DlangTokenType.ID)('stdio')
      PsiElement(DlangTokenType.;)(';')
  DLanguageDeclarationImpl(DECLARATION)
    FUNCTION_DECLARATION
      DLanguageTypeImpl(TYPE)
        DLanguageType_2Impl(TYPE_2)
          DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
            PsiElement(DlangTokenType.void)('void')
      IDENTIFIER
        PsiElement(DlangTokenType.ID)('main')
      DLanguageParametersImpl(PARAMETERS)
        PsiElement(DlangTokenType.()('(')
        PsiElement(DlangTokenType.))(')')
      DLanguageFunctionBodyImpl(FUNCTION_BODY)
        DLanguageSpecifiedFunctionBodyImpl(SPECIFIED_FUNCTION_BODY)
          DLanguageBlockStatementImpl(BLOCK_STATEMENT)
            PsiElement(DlangTokenType.{)('{')
            DLanguageDeclarationsAndStatementsImpl(DECLARATIONS_AND_STATEMENTS)
              DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
                DLanguageDeclarationImpl(DECLARATION)
                  DLanguageVariableDeclarationImpl(VARIABLE_DECLARATION)
                    DLanguageTypeImpl(TYPE)
                      DLanguageType_2Impl(TYPE_2)
                        DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                          PsiElement(DlangTokenType.int)('int')
                    DECLARATOR
                      IDENTIFIER
                        PsiElement(DlangTokenType.ID)('i')
                    PsiElement(DlangTokenType.;)(';')
              DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
                DLanguageStatementImpl(STATEMENT)
                  DLanguageStatementNoCaseNoDefaultImpl(STATEMENT_NO_CASE_NO_DEFAULT)
                    DLanguageExpressionStatementImpl(EXPRESSION_STATEMENT)
                      DLanguageExpressionImpl(EXPRESSION)
                        DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                          DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                            DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                              DLanguageFunctionCallExpressionImpl(FUNCTION_CALL_EXPRESSION)
                                DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                  DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                    DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                      IDENTIFIER
                                        PsiElement(DlangTokenType.ID)('writeln')
                                DLanguageArgumentsImpl(ARGUMENTS)
                                  PsiElement(DlangTokenType.()('(')
                                  DLanguageArgumentListImpl(ARGUMENT_LIST)
                                    DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                      DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                        DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                          DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                            DLanguageTraitsExpressionImpl(TRAITS_EXPRESSION)
                                              PsiElement(DlangTokenType.__traits)('__traits')
                                              PsiElement(DlangTokenType.()('(')
                                              IDENTIFIER
                                                PsiElement(DlangTokenType.ID)('isArithmetic')
                                              PsiElement(DlangTokenType.,)(',')
                                              DLanguageTemplateArgumentListImpl(TEMPLATE_ARGUMENT_LIST)
                                                DLanguageTemplateArgumentImpl(TEMPLATE_ARGUMENT)
                                                  DLanguageTypeImpl(TYPE)
                                                    DLanguageType_2Impl(TYPE_2)
                                                      DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                                                        PsiElement(DlangTokenType.int)('int')
                                              PsiElement(DlangTokenType.))(')')
                                  PsiElement(DlangTokenType.))(')')
                      PsiElement(DlangTokenType.;)(';')
              DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
                DLanguageStatementImpl(STATEMENT)
                  DLanguageStatementNoCaseNoDefaultImpl(STATEMENT_NO_CASE_NO_DEFAULT)
                    DLanguageExpressionStatementImpl(EXPRESSION_STATEMENT)
                      DLanguageExpressionImpl(EXPRESSION)
                        DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                          DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                            DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                              DLanguageFunctionCallExpressionImpl(FUNCTION_CALL_EXPRESSION)
                                DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                  DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                    DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                      IDENTIFIER
                                        PsiElement(DlangTokenType.ID)('writeln')
                                DLanguageArgumentsImpl(ARGUMENTS)
                                  PsiElement(DlangTokenType.()('(')
                                  DLanguageArgumentListImpl(ARGUMENT_LIST)
                                    DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                      DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                        DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                          DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                            DLanguageTraitsExpressionImpl(TRAITS_EXPRESSION)
                                              PsiElement(DlangTokenType.__traits)('__traits')
                                              PsiElement(DlangTokenType.()('(')
                                              IDENTIFIER
                                                PsiElement(DlangTokenType.ID)('isArithmetic')
                                              PsiElement(DlangTokenType.,)(',')
                                              DLanguageTemplateArgumentListImpl(TEMPLATE_ARGUMENT_LIST)
                                                DLanguageTemplateArgumentImpl(TEMPLATE_ARGUMENT)
                                                  DLanguageTypeImpl(TYPE)
                                                    DLanguageType_2Impl(TYPE_2)
                                                      DLanguageTypeIdentifierPartImpl(TYPE_IDENTIFIER_PART)
                                                        DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                                          IDENTIFIER
                                                            PsiElement(DlangTokenType.ID)('i')
                                                PsiElement(DlangTokenType.,)(',')
                                                DLanguageTemplateArgumentImpl(TEMPLATE_ARGUMENT)
                                                  DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                                    DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                                      DLanguageAddExpressionImpl(ADD_EXPRESSION)
                                                        DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                                          DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                                            DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                                              IDENTIFIER
                                                                PsiElement(DlangTokenType.ID)('i')
                                                        PsiElement(DlangTokenType.+)('+')
                                                        DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                                          DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                                            PsiElement(DlangTokenType.INTEGER_LITERAL)('1')
                                                PsiElement(DlangTokenType.,)(',')
                                                DLanguageTemplateArgumentImpl(TEMPLATE_ARGUMENT)
                                                  DLanguageTypeImpl(TYPE)
                                                    DLanguageType_2Impl(TYPE_2)
                                                      DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                                                        PsiElement(DlangTokenType.int)('int')
                                              PsiElement(DlangTokenType.))(')')
                                  PsiElement(DlangTokenType.))(')')
                      PsiElement(DlangTokenType.;)(';')
              DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
                DLanguageStatementImpl(STATEMENT)
                  DLanguageStatementNoCaseNoDefaultImpl(STATEMENT_NO_CASE_NO_DEFAULT)
                    DLanguageExpressionStatementImpl(EXPRESSION_STATEMENT)
                      DLanguageExpressionImpl(EXPRESSION)
                        DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                          DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                            DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                              DLanguageFunctionCallExpressionImpl(FUNCTION_CALL_EXPRESSION)
                                DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                  DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                    DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                      IDENTIFIER
                                        PsiElement(DlangTokenType.ID)('writeln')
                                DLanguageArgumentsImpl(ARGUMENTS)
                                  PsiElement(DlangTokenType.()('(')
                                  DLanguageArgumentListImpl(ARGUMENT_LIST)
                                    DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                      DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                        DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                          DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                            DLanguageTraitsExpressionImpl(TRAITS_EXPRESSION)
                                              PsiElement(DlangTokenType.__traits)('__traits')
                                              PsiElement(DlangTokenType.()('(')
                                              IDENTIFIER
                                                PsiElement(DlangTokenType.ID)('isArithmetic')
                                              PsiElement(DlangTokenType.))(')')
                                  PsiElement(DlangTokenType.))(')')
                      PsiElement(DlangTokenType.;)(';')
              DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
                DLanguageStatementImpl(STATEMENT)
                  DLanguageStatementNoCaseNoDefaultImpl(STATEMENT_NO_CASE_NO_DEFAULT)
                    DLanguageExpressionStatementImpl(EXPRESSION_STATEMENT)
                      DLanguageExpressionImpl(EXPRESSION)
                        DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                          DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                            DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                              DLanguageFunctionCallExpressionImpl(FUNCTION_CALL_EXPRESSION)
                                DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                  DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                    DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                      IDENTIFIER
                                        PsiElement(DlangTokenType.ID)('writeln')
                                DLanguageArgumentsImpl(ARGUMENTS)
                                  PsiElement(DlangTokenType.()('(')
                                  DLanguageArgumentListImpl(ARGUMENT_LIST)
                                    DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                      DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                        DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                          DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                            DLanguageTraitsExpressionImpl(TRAITS_EXPRESSION)
                                              PsiElement(DlangTokenType.__traits)('__traits')
                                              PsiElement(DlangTokenType.()('(')
                                              IDENTIFIER
                                                PsiElement(DlangTokenType.ID)('isArithmetic')
                                              PsiElement(DlangTokenType.,)(',')
                                              DLanguageTemplateArgumentListImpl(TEMPLATE_ARGUMENT_LIST)
                                                DLanguageTemplateArgumentImpl(TEMPLATE_ARGUMENT)
                                                  DLanguageTypeImpl(TYPE)
                                                    DLanguageType_2Impl(TYPE_2)
                                                      DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                                                        PsiElement(DlangTokenType.int)('int')
                                                    DLanguageTypeSuffixImpl(TYPE_SUFFIX)
                                                      PsiElement(DlangTokenType.*)('*')
                                              PsiElement(DlangTokenType.))(')')
                                  PsiElement(DlangTokenType.))(')')
                      PsiElement(DlangTokenType.;)(';')
            PsiElement(DlangTokenType.})('}')